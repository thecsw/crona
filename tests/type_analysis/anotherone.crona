f : int(a : byte) {}
g : byte(b : int, c : bool) {}
h : void() {}

main : void() {
     trashint : int;
     trashbyte : byte;

     trashint = f(true);
     trashint = f(1);
     trashint = f(-1);
     trashint = f(0);
     trashint = f(255);
     trashint = f(256);
     trashint = f(999999);
     trashbyte = g(true, false);
     trashbyte = g(true);
     trashbyte = g(true, true, true);
     trashbyte = g(5, false);
     trashbyte = g(39021, false, 8);
     trashbyte = h();
     trashint = h();
     
     a : int;
     a = f(99) + g(255, true);
     a = h() + g(8, true);
     a = h + g;
     a = f(999) + g(-1, true, false);
     a = f(9999) + f(-1);
     h = g;
     f = g;
     g = f;

     return a;
}