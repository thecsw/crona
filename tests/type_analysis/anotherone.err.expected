FATAL [9,19]: Type of actual does not match type of formal
FATAL [11,19]: Type of actual does not match type of formal
FATAL [14,19]: Type of actual does not match type of formal
FATAL [15,19]: Type of actual does not match type of formal
FATAL [16,20]: Type of actual does not match type of formal
FATAL [17,18]: Function call with wrong number of args
FATAL [18,18]: Function call with wrong number of args
FATAL [20,18]: Function call with wrong number of args
FATAL [21,18]: Invalid assignment operand
FATAL [22,17]: Invalid assignment operand
FATAL [26,10]: Arithmetic operator applied to invalid operand
FATAL [27,10]: Arithmetic operator applied to invalid operand
FATAL [27,14]: Arithmetic operator applied to invalid operand
FATAL [28,12]: Type of actual does not match type of formal
FATAL [28,19]: Function call with wrong number of args
FATAL [29,12]: Type of actual does not match type of formal
FATAL [29,22]: Type of actual does not match type of formal
FATAL [30,6]: Invalid assignment operand
FATAL [30,10]: Invalid assignment operand
FATAL [31,6]: Invalid assignment operand
FATAL [31,10]: Invalid assignment operand
FATAL [32,6]: Invalid assignment operand
FATAL [32,10]: Invalid assignment operand
FATAL [34,13]: Return with a value in void function
Type Analysis Failed
